### ISSUE TITLE
Current cell almost invisible at the bottom

### ISSUE THREAD

**Issue by qwer1304** [spyder-ide/spyder#6415]
## Description of your problem

**What steps will reproduce the problem?**

1. Have a source with several cells
2. Use Next cell or Run cell and advance commands several times
3. 

**What is the expected output? What do you see instead?**
The current cell should be well visible (preferably, at the middle of screen)

The current cell is at the bottom of screen, with only two lines visible, so every time I need to press Down arrow to make the cell visible

**Please provide any additional information below**

## Versions and main components

* Spyder Version: 3.2.6
* Python Version: 3.5
* Qt Version: 5.6.2
* PyQt Version: 5.6.0
* Operating system: Win 10 64-bit

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:
IPython >=4.0    :  6.2.1 (OK)
cython >=0.21    :  0.27.3 (OK)
jedi >=0.9.0     :  0.11.1 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
numpy >=1.7      :  1.14.0 (OK)
pandas >=0.13.1  :  0.22.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.6.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.8.2 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.10.7 (OK)
sphinx >=0.6.6   :  1.6.6 (OK)
sympy >=0.7.3    :  1.1.1 (OK)

**Comment by ccordoba12** [comment]
Thanks for reporting. This is a very good suggestion, we'll try to implement it in a future release.

**Comment by CAM-Gerlach** [comment]
Indeed, confirmed. I would suggest scrolling to at least the last line of the selected cell, or maybe one or so beyond (to see any title of the next).

**Comment by qwer1304** [comment]
I'd rather see the top of the selected cell to get some context. So maybe put the top of the selected cell in the middle of the screen?

**Comment by CAM-Gerlach** [comment]
Well, you would see it. At least from my perspective, since the point of the command is moving to the next cell, I'd think the focus ought to be on showing the whole cell (supposing it fits on the screen, if not, then of course you'd just see the top n lines that fit.), rather than moving it to a specific, arbitrary position, regardless of length. On the other hand, I'd suppose an advantage of the "middle of the screen" approach would be putting a certain part of the cell (top? center?) in a fixed position when moving, and might show a little more balanced context of the other cells around it. 

However, the question would then be whether the center of the cell should be matched to the center of the screen to make it even, but possibly cut off the top (and have to calculate where the center is), or the top line of the cell, in which case it would be unbalanced and could only show half the total cell length that the "whole cell if possible" approach could. Interesting decision.

**Comment by qwer1304** [comment]
If the whole cell fits the screen then IMHO it doesn't matter where you put it as long as it's visible.
If not - I'd put the **top** of the cell in the **middle** and be done.
So actually, putting the **top** of the cell in the **middle** of the screen would do this for me.

**Comment by CAM-Gerlach** [comment]
Its not a huge deal either way, but while your point is certainly noted, and I do cite a few potential advantages of that arrangement above, I'd be curious to actually hear your reasoning for that vs. showing the whole cell the user selected on screen plus an extra line or two at the bottom, extending the top of the cell to the top of the screen if pessary (but no further), with the major advantage of being able to use screen space up to twice as efficiently to show more or all of the cell the user selected, vs. effectively "wasting" half of it—I'm not clear on exactly what you see as the main gains in return in this scenario.

**Comment by qwer1304** [comment]
To me, the important part is:
1.see the top of the cell
2.see the maximum amount of the cell

The rest would be dictated by how much free space is left thereafter

**Comment by CAM-Gerlach** [comment]
Thanks very much for clarifying. That, then would in fact dictate the approach I suggested above: Align the bottom of the cell to the bottom of the page, plus a few extra lines (maximizing the amount viewed), unless the cell is larger than the height of the editor frame, in which case align the top of the cell to the top of the window. So:

Small Cell:

![image](https://user-images.githubusercontent.com/17051931/36065770-21ca6736-0e65-11e8-8a89-5c19ea6633e3.png)

Big Cell:

![image](https://user-images.githubusercontent.com/17051931/36065744-bf0723f0-0e64-11e8-96f9-d4b12547b31a.png)

Bigger than screen cell:

![image](https://user-images.githubusercontent.com/17051931/36065762-f8acd118-0e64-11e8-858a-952f3ae95662.png)
